ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -271,8 +308,9 @@ private PullQueryResult handlePullQuery(
   ) {
     // Get active and standby nodes for this key
     final Locator locator = pullQueryContext.mat.locator();
+    final Struct key = Iterables.getLast(pullQueryContext.keys);
     final List<KsqlNode> filteredAndOrderedNodes = locator.locate(
-        pullQueryContext.key,
+        key,
```

Comment:
You already did the locate when splitting the keys into groups

Reply:
I had thought that a given set of partitions were grouped together at each active and standby, but I think you're right this isn't the case.  I'll rework this.